common/windows/win32_window.h

```diff
@@ -0,0 +1,92 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef WIN32_WINDOW_H_
+#define WIN32_WINDOW_H_
+
+#include <Windows.h>
+#include <Windowsx.h>
+
+#include <memory>
+#include <string>
+
+// A class abstraction for a high DPI aware Win32 Window.  Intended to be
+// inherited from by classes that wish to specialize with custom
+// rendering and input handling
+class Win32Window {
```

Comment:
How would setting title, icon, size, etc. work here, given that the handle is internal?

Reply:
Handle exposed in next commit